### ISSUE TITLE
Error create Foreign key

### ISSUE THREAD

**Issue by dsivokhin** [directus/directus#5999]
When I create translations field

![изображение](https://user-images.githubusercontent.com/4669992/120494157-4004e100-c3c4-11eb-80c6-dc318b413e52.png)

Steps: 
1)Create some table
2)Try create translations field

**new table**
![изображение](https://user-images.githubusercontent.com/4669992/120494959-eea92180-c3c4-11eb-96a2-cbc9ed71efe4.png)

**languages table**
![изображение](https://user-images.githubusercontent.com/4669992/120495154-1dbf9300-c3c5-11eb-8042-85b85eb6a98a.png)

As I understand it, this error is because the fields are not identical. 
INT(10) - by default when type UNSIGNED
INT(11) - by default when not  UNSIGNED

When I manually make the types identical (INT(10) and UNSIGNED both), there is no error.

Directus 9.0.0-rc70,  
MySQL 5.7.33

**Comment by RSickenberg** [comment]
Got the same issue as well!

**Comment by TorbenHammes** [comment]
I tried it with RC71 and I have the same problem.

**Comment by rijkvanzanten** [comment]
@RSickenberg @TorbenHammes Both on MySQL 5.x as well? Or on a different database?

**Comment by TorbenHammes** [comment]
I use MariaDB 10.3.29 on Ubuntu 20

**Comment by RSickenberg** [comment]
@rijkvanzanten I use `mariadb:latest` image.

**Comment by Mrmiffo** [comment]
I have the same issue on rc73 as well

**Comment by rijkvanzanten** [comment]
@Mrmiffo What exact version of MySQL / MariaDB, and what is the exact error?

**Comment by Mrmiffo** [comment]
> @Mrmiffo What exact version of MySQL / MariaDB, and what is the exact error?

MySql 8, I'll check the error on Monday!

**Comment by rijkvanzanten** [comment]
Thanks! Also make sure that all your tables/columns are using the correct collation/character set. I've had some MySQL headaches before from different tools using different defaults in their connections :|

**Comment by Mrmiffo** [comment]
Sorry for the late reply! 

I tried this using rc74 today in a clean environment today, but I'm unable to recreate it there. But I still get the error in prod: 
```
{
  "errors": [
    {
      "message": "alter table `test_translations` add constraint `test_translations_languages_code_foreign` foreign key (`languages_code`) references `languages` (`code`) on delete SET NULL - UNKNOWN_CODE_PLEASE_REPORT: Referencing column 'languages_code' and referenced column 'code' in foreign key constraint 'test_translations_languages_code_foreign' are incompatible.",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

Stack trace:
```
09:13:43 ­ƒÜ¿ alter table `test_translations` add constraint `test_translations_languages_code_foreign` foreign key (`languages_code`) references `languages` (`code`) on delete SET NULL - UNKNOWN_CODE_PLEASE_REPORT: Referencing column 'languages_code' and referenced column 'code' in foreign key constraint 'test_translations_languages_code_foreign' are incompatible. 
Error: UNKNOWN_CODE_PLEASE_REPORT: Referencing column 'languages_code' and referenced column 'code' in foreign key constraint 'test_translations_languages_code_foreign' are incompatible.
    at Query.Sequence._packetToError (D:\oobcms-v9\node_modules\mysql\lib\protocol\sequences\Sequence.js:47:14)
    at Query.ErrorPacket (D:\oobcms-v9\node_modules\mysql\lib\protocol\sequences\Query.js:79:18)
    at Protocol._parsePacket (D:\oobcms-v9\node_modules\mysql\lib\protocol\Protocol.js:291:23)
    at Parser._parsePacket (D:\oobcms-v9\node_modules\mysql\lib\protocol\Parser.js:433:10)
    at Parser.write (D:\oobcms-v9\node_modules\mysql\lib\protocol\Parser.js:43:10)
    at Protocol.write (D:\oobcms-v9\node_modules\mysql\lib\protocol\Protocol.js:38:16)
    at Socket.<anonymous> (D:\oobcms-v9\node_modules\mysql\lib\Connection.js:88:28)
    at Socket.<anonymous> (D:\oobcms-v9\node_modules\mysql\lib\Connection.js:526:10)
    at Socket.emit (events.js:315:20)
    at addChunk (_stream_readable.js:309:12)
    --------------------
    at Protocol._enqueue (D:\oobcms-v9\node_modules\mysql\lib\protocol\Protocol.js:144:48)
    at Connection.query (D:\oobcms-v9\node_modules\mysql\lib\Connection.js:198:25)
    at D:\oobcms-v9\node_modules\knex\lib\dialects\mysql\index.js:126:18
    at new Promise (<anonymous>)
    at Client_MySQL._query (D:\oobcms-v9\node_modules\knex\lib\dialects\mysql\index.js:120:12)
    at executeQuery (D:\oobcms-v9\node_modules\knex\lib\execution\internal\query-executioner.js:37:17)
    at Client_MySQL.query (D:\oobcms-v9\node_modules\knex\lib\client.js:134:12)
    at D:\oobcms-v9\node_modules\knex\lib\execution\transaction.js:363:24
    at new Promise (<anonymous>)
    at Client_MySQL.trxClient.query (D:\oobcms-v9\node_modules\knex\lib\execution\transaction.js:358:12)
```

I do notice some interesting things about the collations. In my existing translations, the language column has the collation "utf8_general_ci", which is the same as the collation of "code" in the languages table. But if I try to create a new translation, the collation of the language column in the new translation is "utf8mb4_0900_ai_ci". Which explains why it works in a clean environment as in that environment the languages table does not exist when the translation is created, so directus creates it with the same collation.

**Comment by rijkvanzanten** [comment]
@Mrmiffo 

> NOTE: All tables and columns have been created using Directus, so it would be Directus that has changed which collation it is using between version, and is not backward compatible with itself.

Directus itself doesn't set/alter collation/character sets at all. These differences in type must've been caused by an update to the database itself (new default), changes in the connection configuration (in env vars), or from migrating a installation between different databases 🤔

**Comment by Mrmiffo** [comment]
I checked some of our later tables today, and indeed these have another collation than our original tables. As such the rc.70+ upgrade was not the culprit but merely the messenger :) I'll modify all our table to have the same collation and the issue should be resolved.

My guess is that, at some point, the MySQL connector installed along with Directus has been updated at some point, resulting in the use of a different collation. But as we Directus did not use foreign key constraints until now, this has gone unnoticed.
